src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -488,7 +488,7 @@ bool usesOriginalInput(BoundDecisionDagNode node)
                 bool canShareInputs,
                 out BoundExpression savedInputExpression)
             {
-                int count = loweredInput.Arguments.Length;
+                int count = loweredInput.Type.TupleElements.Length;
```

Comment:
For tuple containing > 7 elements (10 elements for example), the previous value of count will be 8 (the TRest position) instead of 10, causing the loop after a few lines to run a fewer iteration than needed.